src/app/search-results/search-results.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnDestroy, OnChanges, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Channel } from '../shared/models/channel.model';
+import { Video } from '../shared/models/video.model';
+import { ChannelService } from '../core/services/channel.service';
+import { VideoService } from '../core/services/video.service';
+
+@Component({
+  selector: 'bs-search-results',
+  templateUrl: './search-results.component.html',
+  styleUrls: ['./search-results.component.scss']
+})
+export class SearchResultsComponent implements OnDestroy, OnInit {
+
+  constructor(
+    private route: ActivatedRoute,
+    private channelService: ChannelService,
+    private videoService: VideoService
+    ) { }
+
+  routeQuerySub: any;
```

Comment:
It's better to call the variable `routeQuerySubscription` instead of `routeQuerySub`,
same for `videosSub` and `channelsSub`

Reply:
Changed